test/test_dependency_versions.py

```diff
@@ -111,7 +111,6 @@ def test_dependency_constraints_file(tmp_path, build_frontend_env):
             """
             pip=={pip}
             delocate=={delocate}
-            importlib-metadata<3,>=0.12; python_version < "3.8"
```

Comment:
Failing tests on windows came from this pin.
I'm not sure exactly why since I can't really debug on Windows.
All I can say is that it was causing some backtracking in the pip resolver in order to install an old version of virtualenv (2.16.2).